clicommand/agent_start.go

```diff
@@ -692,6 +693,15 @@ var AgentStartCommand = cli.Command{
 			l.Fatal("Failed to parse cancel-signal: %v", err)
 		}
 
+		// confirm the BuildPath is exists. The bootstrap is going to write to it when a job executes,
+		// so we may as well check that'll work now and fail early if it's a problem
+		if !utils.FileExists(agentConf.BuildPath) {
+			l.Info("Build Path doesn't exist, creating it (%s)", agentConf.BuildPath)
+			if err := os.MkdirAll(agentConf.BuildPath, 0777); err != nil {
+				l.Fatal("Failed to create builds path: %v", err)
+			}
+		}
```

Comment:
Is this a controversial thing to do? The bootstrap process will implicitly create the builds directory later on, but it might be worth doing this explicitly in the main agent process and failing early?

Here's where it's done in the bootstrap: https://github.com/buildkite/agent/blob/fce406bd64a0d5a6ba2075931bbfd99cc2aeca65/bootstrap/bootstrap.go#L956-L961

This is include in what was going to be a 1-line PR because the bk cli integration specs specify a BuildPath that doesn't exist, and we're unable to start the bootstrap processes with a working directory that doesn't exist.

Reply:
Is `os.MkdirAll(checkoutPath, 0777)` legit on Windows/NTFS?
I imagine so? Is this the same code we would have run later on in the bootstrap anyway? Or different?

Reply:
Looks like the shape of the code is the same no matter the OS:

https://golang.org/src/os/file.go?s=7592:7636#L245

Reply:
> Is this the same code we would have run later on in the bootstrap anyway? Or different?

It's the same code we run in the bootstrap later to create the checkout directory, which implicitly creates the builds directory as well (if it doesn't exist).